Source/Core/Common/MemArenaWin.cpp

```diff
@@ -441,22 +443,79 @@ LazyMemoryRegion::~LazyMemoryRegion()
   Release();
 }
 
+// On Windows, it's not possible to tell the OS to allocate a memory region without reserving space
+// for it in either physical memory or the pagefile. This unfortunately leads to the pagefile
+// growing by up to 32 GB whenever the new JIT entry point lookup table optimization is used (see
+// m_entry_points_ptr in JitBaseBlockCache). So as a workaround, we only reserve the memory space
+// for it, and actually allocate the individual pages on first access via a segfault handler.
+
+// FIXME: This allows only a single LazyMemoryRegion to exist in the entire process at any given
+// point in time because there is no good way to pass userdata to an exception handler. As of
+// writing this isn't really a problem -- fastmem has the same problem, in fact -- but it might
+// become one in the future. This construct needs to be refectored in that case.
+
+LazyMemoryRegion* s_lazy_region = nullptr;
+
+static LONG NTAPI LazyMemoryRegionExceptionHandler(PEXCEPTION_POINTERS pPtrs)
+{
+  if (pPtrs->ExceptionRecord->ExceptionCode != EXCEPTION_ACCESS_VIOLATION)
+    return EXCEPTION_CONTINUE_SEARCH;
+
+  const ULONG_PTR access_type = pPtrs->ExceptionRecord->ExceptionInformation[0];
+  if (access_type == 8)  // Rule out DEP
+    return EXCEPTION_CONTINUE_SEARCH;
```

Comment:
It's probably worth mentioning that 8 is a magic number given directly on https://learn.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-exception_record (for both `EXCEPTION_ACCESS_VIOLATION` and `EXCEPTION_IN_PAGE_ERROR`) and I don't think there's any constant for it.

Reply:
afaict it's specified as a magic number in some places in the documentation, and as "bit 0 set means write, bit 3 set means execute" [in other places like bugcheck documentation](https://learn.microsoft.com/en-us/windows-hardware/drivers/debugger/bug-check-0xa--irql-not-less-or-equal#irql_not_less_or_equal-parameters).